include/dsn/utility/strings.h

```diff
@@ -49,5 +50,23 @@ char *trim_string(char *s);
 
 // calculate the md5 checksum of buffer
 std::string string_md5(const char *buffer, unsigned int length);
+
+// merge strings with specified delimiter
+template <typename ForwardIterator>
+std::string merge(ForwardIterator begin, ForwardIterator end, const std::string &delimiter)
```

Comment:
I think [boost::join](https://www.boost.org/doc/libs/1_65_1/boost/algorithm/string/join.hpp) can replace this function.